core/src/main/java/kafka/server/share/SharePartitionManager.java

```diff
@@ -665,17 +670,78 @@ public void handleFencedSharePartitionException(
             // The share partition is fenced hence remove the partition from map and let the client retry.
             // But surface the error to the client so client might take some action i.e. re-fetch
             // the metadata and retry the fetch on new leader.
-            SharePartition sharePartition = partitionCacheMap.remove(sharePartitionKey);
-            if (sharePartition != null) {
-                sharePartition.markFenced();
-            }
+            removeSharePartitionFromCache(sharePartitionKey, partitionCacheMap, replicaManager);
         }
     }
 
     private SharePartitionKey sharePartitionKey(String groupId, TopicIdPartition topicIdPartition) {
         return new SharePartitionKey(groupId, topicIdPartition);
     }
 
+    private static void removeSharePartitionFromCache(SharePartitionKey sharePartitionKey,
+        Map<SharePartitionKey, SharePartition> map, ReplicaManager replicaManager) {
+        SharePartition sharePartition = map.remove(sharePartitionKey);
```

Comment:
Nit: A line break before the ) would separate the argument declarations from the statements more clearly.

Reply:
Done.